### PR TITLE
Chore: View form responses as user messages

### PR DESCRIPTION
**Pull request by pranavrajs**
- View user response as form messages

### Form

Screenshot 2020-05-17 21 54 45

### After response is submitted 

Screenshot 2020-05-17 21 55 28

### CODE REVIEW

File: app/javascript/widget/components/AgentMessage.vue [hunk 1]
```diff
@@ -42,6 +42,13 @@
     </div>
 
     <UserMessage v-if="hasRecordedResponse" :message="responseMessage" />
+    <div v-if="isASubmittedForm">
```
Comment (nithindavid) [inline app/javascript/widget/components/AgentMessage.vue:45]: Is this div necessary ? css `+` rules might not catch it if it's inside div.

File: app/javascript/widget/store/modules/conversation.js [hunk 2]
```diff
@@ -61,15 +61,22 @@ export const getters = {
     return Object.keys(conversationGroupedByDate).map(date => {
       const messages = conversationGroupedByDate[date].map((message, index) => {
         let showAvatar = false;
-        if (index === conversationGroupedByDate[date].length - 1) {
+        if (
+          message.content_type === 'form' &&
```
Comment (nithindavid) [inline app/javascript/widget/store/modules/conversation.js:65]: Make this check a variable for better readability ?

File: app/javascript/widget/store/modules/conversation.js [hunk 3]
```diff
@@ -61,15 +61,22 @@ export const getters = {
...
           showAvatar = true;
         } else {
           const nextMessage = conversationGroupedByDate[date][index + 1];
           const currentSender = message.sender ? message.sender.name : '';
           const nextSender = nextMessage.sender ? nextMessage.sender.name : '';
           showAvatar =
             currentSender !== nextSender ||
-            message.message_type !== nextMessage.message_type;
+            message.message_type !== nextMessage.message_type ||
+            (nextMessage.content_type === 'form' &&
```
Comment (nithindavid) [inline app/javascript/widget/store/modules/conversation.js:78]: Here also a variable with good name

File: app/javascript/widget/components/AgentMessage.vue [hunk 4]
```diff
@@ -147,6 +154,20 @@ export default {
...
       return '';
     },
+    isASubmittedForm() {
+      return (
+        this.contentType === 'form' &&
+        this.messageContentAttributes.submitted_values
+      );
+    },
+    submittedFormValues() {
+      return this.messageContentAttributes.submitted_values.map(
```
Comment (nithindavid) [inline app/javascript/widget/components/AgentMessage.vue:164]: prefer to destruct and use snake cases into camel cases

File: app/javascript/widget/store/modules/conversation.js (hunk 3 above)
Comment (pranavrajs) [reply app/javascript/widget/store/modules/conversation.js]: Fixed.

File: app/javascript/widget/store/modules/conversation.js (hunk 2 above)
Comment (pranavrajs) [reply app/javascript/widget/store/modules/conversation.js]: Fixed.

**Pull request merged**